pkg/pool-stable/test/StablePool.test.ts

```diff
@@ -694,21 +799,33 @@ describe('StablePool', function () {
           });
 
           itPaysExpectedProtocolFees();
+
+          context('still works in recovery mode', () => {
+            sharedBeforeEach('enter recovery mode', async () => {
+              await enterRecoveryMode(pool);
+            });
+
+            itPaysExpectedProtocolFees();
+          });
         });
 
         function itPaysExpectedProtocolFees() {
-          it('pays swap protocol fees on join exact tokens in for BPT out', async () => {
+          it('pays swap protocol fees on join exact tokens in for BPT out (unless in recovery)', async () => {
             const result = await pool.joinGivenIn({
               from: lp,
               amountsIn: fp(1),
               currentBalances,
               protocolFeePercentage,
             });
 
-            expect(result.dueProtocolFeeAmounts).to.be.equalWithError(expectedDueProtocolFeeAmounts, 0.0001);
+            if (await pool.inRecoveryMode()) {
+              expect(result.dueProtocolFeeAmounts).to.be.zeros;
+            } else {
+              expect(result.dueProtocolFeeAmounts).to.be.equalWithError(expectedDueProtocolFeeAmounts, 0.0001);
+            }
```

Comment:
This is also out of scope for testing, as BasePool will simply report a fee % of 0.